### PR TITLE
Implemented Gradle Build System

### LINKED ISSUE #39: Implement Build Process using Gradle
Implement Build Process using Gradle
 
 - [x] Should build the whole app
 
#
 - Primary: @JonathanMathBoi
 - Secondary: @JJSnader 
 - Size: M
 - Actual Size: (fib number after completion)
 - Actual Time: (in minutes or hours -- after completion)

### PR DESCRIPTION
**Pull request by JonathanMathBoi**
Added the Gradle Build System and reorganized the file structure to a standard Java project.

Resolves #39

### CODE REVIEW

File: app/src/main/java/xterminators/spellingbee/UserFunctions.java [hunk 1]
```diff
@@ -18,9 +20,13 @@ public UserFunctions() {
     }
 
     /** The path to the dictionary file for the game. */
-    public static final String DICTIONARY_PATH = "src\\dictionary_optimized.txt";
+    public static final String DICTIONARY_PATH
+        = "app" + File.separator + "src" + File.separator + "main" +
```
Comment (JJSnader) [inline app/src/main/java/xterminators/spellingbee/UserFunctions.java:24]: Maybe we should put File.separator in a constant with a short name just to improve readability. Not a code problem, but having "File.separator" all over the place makes it harder to read.

**Review by JJSnader — APPROVED** [review]
I think this looks fine. I am not really familiar with Gradle at all, but the rearranged file structure looks good.

File: app/src/main/java/xterminators/spellingbee/UserFunctions.java (hunk 1 above)
Comment (JonathanMathBoi) [reply app/src/main/java/xterminators/spellingbee/UserFunctions.java:24]: How would you feel about something like `Paths.get("app", "src", "main", "resources", "dictionary_optimized.txt").toString()`, which would return the same thing? That way we don't we don't need a short name variable that would add its own readability concerns.

Comment (JJSnader) [reply app/src/main/java/xterminators/spellingbee/UserFunctions.java:24]: Yeah, that would probably be the best option. A string with a bunch of concatenation is always confusing to read.

Comment (JonathanMathBoi) [reply app/src/main/java/xterminators/spellingbee/UserFunctions.java]: Added this in commit 68c185ddbc3f7e29f1048afb22967308f8bebd7d.

**Review by JJSnader — APPROVED** [review]
Looks good.

**Pull request merged**